pieces/pieces.go

```diff
@@ -17,3 +20,62 @@ var Display = map[string]string{
 	"q": "♕",
 	"r": "♖",
 }
+
+// IsWhite returns true if the piece is white.
+func (p Piece) IsWhite() bool {
+	return p == "P" || p == "R" || p == "N" || p == "B" || p == "Q" || p == "K"
+}
+
+// IsBlack returns true if the piece is black.
+func (p Piece) IsBlack() bool {
+	return p == "p" || p == "r" || p == "n" || p == "b" || p == "q" || p == "k"
+}
```

Comment:
I think these functions could be cleaned up a bit.
```suggestion
// IsWhite returns true if the piece is white.
func (p Piece) IsWhite() bool {
    return strings.ToLower(p) != p
}

// IsBlack returns true if the piece is black.
func (p Piece) IsBlack() bool {
    return strings.ToLower(p) == p
}
```

Alternatively, we could also check whether the ASCII code is within a range.